pypechain/render/contract.py

```diff
@@ -19,14 +23,98 @@
     get_structs_for_abi,
     is_abi_event,
     is_abi_function,
-    load_abi_from_file,
 )
 from pypechain.utilities.format import capitalize_first_letter_only
 from pypechain.utilities.templates import get_jinja_env
-from pypechain.utilities.types import EventData, FunctionData, SignatureData, gather_matching_types
+from pypechain.utilities.types import EventData, FunctionData, SignatureData
 
 
-def render_contract_file(contract_name: str, abi_file_path: Path) -> str:
+@dataclass
+class ContractInfo:
+    """Contract Information."""
+
+    abi: ABI
+    bytecode: str
+    contract_name: str
+    structs: dict[str, StructInfo]
+    events: dict[str, EventInfo]
+
+
+def get_contract_infos(abi_infos: list[AbiInfo]) -> dict[str, ContractInfo]:
+    """Get a dictionary of ContractInfos from an AbiInfos list.
+
+     This helps us organize structs and events by the contracts they are defined
+     in. Because structs can be defined in one file, and used in another, we
+     need to avoid duplication of the definitions.
+
+    Parameters
+    ----------
+    abi_infos : list[AbiInfo]
+        _description_
```

Comment:
might as well just fill these in now, it's only going to get harder to do it later